test/src/test/java/hudson/cli/PlainCLIProtocolTest.java

```diff
@@ -0,0 +1,61 @@
+package hudson.cli;
+
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasItem;
+
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+import java.util.logging.Level;
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.jvnet.hudson.test.LoggerRule;
+
+public class PlainCLIProtocolTest {
+
+    @Rule public LoggerRule logger = new LoggerRule().record(PlainCLIProtocol.class, Level.FINE).capture(50);
+
+    @Test
+    public void streamCorruption() throws Exception {
+        final PipedOutputStream upload = new PipedOutputStream();
+        final PipedOutputStream download = new PipedOutputStream();
+
+        class Server extends PlainCLIProtocol.ServerSide {
```

Comment:
I originally was going to put this in the CLI module's unit tests, but without `LoggerRule` I had no way to write any assertions related to the `FlightRecorderInputStream` diagnostics.